src/components/TodoItem.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+    <div class="todo-item">
+        <form>
+            <div>
+                <label for="title">Title: </label>
+                <input type="text" name="title" id="title" v-model="title" :readonly='edit'>
+            </div>
+            <div>
+                <input class="item-desc" type="text" name="description" id="description" :readonly='edit' v-model="description">
+            </div>
+            <div>
+                <label for="pririty">Priority: </label>
+                <input type="text" name="priority"  id="priority" :readonly='edit' v-model="priority">
+            </div>
+            <div>
+                <label for="description">Complete: </label>
+                <input type="checkbox" name="completed" id="completed" v-model="completed">
+            </div>
+            <button>Edit</button>
+            <button>Delete</button>
+        </form>
+    </div>
+</template>
+
+<script>
+export default {
+    props:{
+        todo: {}
+    },
+    data() {
+        return{
+            title: '',
+            description: '',
+            priority: '',
+            completed: '',
+            edit: true
+        }
+    },
+    mounted(){
+        this.title = this.todo.title
+        this.description = this.todo.description
+        this.priority = this.todo.priority
+        this.completed = this.todo.completed
```

Comment:
Ovo moze na mnogo bolje nacin da odradis: 
kreiras objekat      
data() {
        return{
           posts: {
               title: '',
               description: '',
               priority: '',
                completed: '',
                edit: true
                }
        }
    }

I onda u monted samo response mozes da kaze this.posts { ...this.todo}